torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -298,7 +289,9 @@ def __init__(
 
         edge_type, edge_label_index = get_edge_label_index(
             data, edge_label_index)
-
+        if edge_label is None:
+            edge_label = edge_label_index.new_zeros(edge_label_index.size(1))
+        self.edge_label = edge_label.to(torch.float)
```

Comment:
```suggestion
            edge_label = torch.new_zeros(edge_label_index.size(1), device=edge_label_index.device)
        self.edge_label = edge_label
```

Reply:
if the user passes a `torch.long` `edge_label` we need to convert to `torch.float`.